esp_isotp/src/esp_isotp.c

```diff
@@ -4,88 +4,280 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 #include <stdarg.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/queue.h>
+#include "freertos/FreeRTOS.h"
+#include "esp_check.h"
 #include "esp_log.h"
 #include "esp_timer.h"
 #include "esp_twai.h"
-#include "esp_check.h"
 #include "esp_isotp.h"
-// Ensure configuration is included before isotp.h
 #include "isotp_config.h"
 #include "isotp.h"
 
 static const char *TAG = "esp_isotp";
 
 /**
- * @brief ISO-TP link structure
+ * @brief Determine if the given ID requires extended (29-bit) format
+ *
+ * @param id TWAI identifier to check
+ * @return true if ID requires 29-bit extended format, false for 11-bit standard format
+ */
+static inline bool is_extended_id(uint32_t id)
+{
+    return (id > TWAI_STD_ID_MASK);
```

Comment:
We can't stop users from using 29-digit IDs to express small numbers like 0x100, although 11 digits is enough